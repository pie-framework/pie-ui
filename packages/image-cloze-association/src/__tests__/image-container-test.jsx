import React from 'react';
import _ from 'lodash';
import { shallow } from 'enzyme';

import ImageContainer from '../image-container';

const answer1 = { id: 1, value: '1', containerIndex: 0 };
const answer2 = { id: 2, value: '2', containerIndex: 1 };

const container1 = {
  x: 0,
  y: 0,
  height: '0%',
  width: '0%'
};
const container2 = {
  x: 1,
  y: 1,
  height: '1%',
  width: '1%'
};

describe('Image Container', () => {
  let wrapper;

  const mkWrapper = (opts = {}) => {
    opts = _.extend(
      {
        answers: [],
        canDrag: true,
        draggingElement: {},
        image: {},
        onAnswerSelect: () => {},
        onDragAnswerBegin: () => {},
        onDragAnswerEnd: () => {},
        responseContainers: []
      },
      opts
    );

    return shallow(<ImageContainer {...opts} />);
  };

  beforeEach(() => {
    wrapper = mkWrapper();
  });

  describe('snapshots', () => {
    describe('answers', () => {
      it('renders', () => {
        const wrapper = mkWrapper([answer1, answer2]);
        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('canDrag', () => {
      it('renders', () => {
        const wrapper = mkWrapper({ canDrag: false });
        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('draggingElement', () => {
      it('renders', () => {
        const wrapper = mkWrapper({ draggingElement: answer2 });
        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('image', () => {
      it('renders', () => {
        const wrapper = mkWrapper({
          src: 'https://picsum.photos/id/102/200/300',
          width: 0,
          scale: false,
          height: 0
        });
        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('responseContainers', () => {
      it('renders', () => {
        const wrapper = mkWrapper([container1, container2]);
        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});
