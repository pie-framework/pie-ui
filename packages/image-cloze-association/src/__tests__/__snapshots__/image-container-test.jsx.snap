// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Container snapshots answers renders 1`] = `
<ImageContainer
  0={
    Object {
      "containerIndex": 0,
      "id": 1,
      "value": "1",
    }
  }
  1={
    Object {
      "containerIndex": 1,
      "id": 2,
      "value": "2",
    }
  }
  answers={Array []}
  canDrag={true}
  classes={
    Object {
      "base": "ImageContainer-base-1",
    }
  }
  draggingElement={Object {}}
  duplicateResponses={false}
  image={Object {}}
  onAnswerSelect={[Function]}
  onDragAnswerBegin={[Function]}
  onDragAnswerEnd={[Function]}
  responseContainers={Array []}
/>
`;

exports[`Image Container snapshots canDrag renders 1`] = `
<ImageContainer
  answers={Array []}
  canDrag={false}
  classes={
    Object {
      "base": "ImageContainer-base-1",
    }
  }
  draggingElement={Object {}}
  duplicateResponses={false}
  image={Object {}}
  onAnswerSelect={[Function]}
  onDragAnswerBegin={[Function]}
  onDragAnswerEnd={[Function]}
  responseContainers={Array []}
/>
`;

exports[`Image Container snapshots draggingElement renders 1`] = `
<ImageContainer
  answers={Array []}
  canDrag={true}
  classes={
    Object {
      "base": "ImageContainer-base-1",
    }
  }
  draggingElement={
    Object {
      "containerIndex": 1,
      "id": 2,
      "value": "2",
    }
  }
  duplicateResponses={false}
  image={Object {}}
  onAnswerSelect={[Function]}
  onDragAnswerBegin={[Function]}
  onDragAnswerEnd={[Function]}
  responseContainers={Array []}
/>
`;

exports[`Image Container snapshots image renders 1`] = `
<ImageContainer
  answers={Array []}
  canDrag={true}
  classes={
    Object {
      "base": "ImageContainer-base-1",
    }
  }
  draggingElement={Object {}}
  duplicateResponses={false}
  height={0}
  image={Object {}}
  onAnswerSelect={[Function]}
  onDragAnswerBegin={[Function]}
  onDragAnswerEnd={[Function]}
  responseContainers={Array []}
  scale={false}
  src="https://picsum.photos/id/102/200/300"
  width={0}
/>
`;

exports[`Image Container snapshots responseContainers renders 1`] = `
<ImageContainer
  0={
    Object {
      "height": "0%",
      "width": "0%",
      "x": 0,
      "y": 0,
    }
  }
  1={
    Object {
      "height": "1%",
      "width": "1%",
      "x": 1,
      "y": 1,
    }
  }
  answers={Array []}
  canDrag={true}
  classes={
    Object {
      "base": "ImageContainer-base-1",
    }
  }
  draggingElement={Object {}}
  duplicateResponses={false}
  image={Object {}}
  onAnswerSelect={[Function]}
  onDragAnswerBegin={[Function]}
  onDragAnswerEnd={[Function]}
  responseContainers={Array []}
/>
`;
